### ISSUE TITLE
RangeBitmap can give wrong result when computing between on 2 values with different highest bit

### ISSUE THREAD

**Issue by Jackie-Jiang** [RoaringBitmap/RoaringBitmap#586]
**Describe the bug**
When setting the RangeBitmap to full range (`0xFFFFFFFFFFFFFFFFL`, used for computing double range), computing a between for 2 values with different highest bit won't return the expected matching value.

**To Reproduce**
```
  public static void main(String[] args) {
    RangeBitmap.Appender appender = RangeBitmap.appender(0xFFFFFFFFFFFFFFFFL);
    appender.add(0xFFFFFFFFFFFFFFF0L);
    RangeBitmap rangeBitmap = appender.build();
    System.out.println(rangeBitmap.between(0x0FFFFFFFFFFFFFFFL, 0xFFFFFFFFFFFFFFF0L));
    System.out.println(rangeBitmap.between(0x0FFFFFFFFFFFFFFFL, 0xFFFFFFFFFFFFFFF1L));
  }
```

It returns empty bitmap in both operations.

**RoaringBitmap version:**
0.9.28

**Java version:**
11

**Comment by richardstartin** [comment]
This should be easy to fix. As a quick workaround you can replace between with the following:

```java
    // prints {}
    System.out.println(rangeBitmap.between(0x0FFFFFFFFFFFFFFFL, 0xFFFFFFFFFFFFFFF0L));
    // prints {0}
    System.out.println(rangeBitmap.gte(0x0FFFFFFFFFFFFFFFL, rangeBitmap.lte(0xFFFFFFFFFFFFFFF0L)));
    // prints {}
    System.out.println(rangeBitmap.between(0x0FFFFFFFFFFFFFFFL, 0xFFFFFFFFFFFFFFF1L));
    // prints {0}
    System.out.println(rangeBitmap.gte(0x0FFFFFFFFFFFFFFFL, rangeBitmap.lte(0xFFFFFFFFFFFFFFF1L)));
```

**Comment by Jackie-Jiang** [comment]
@richardstartin Thanks for the quick response! I'll try the workaround

**Comment by richardstartin** [comment]
Hi @Jackie-Jiang, the bug is fixed by #587, I'll submit a fix to PR once it's released.
